### ISSUE TITLE
Polyfill is broken when used with nested popovers and inside shadow DOM

### ISSUE THREAD

**Issue by jpzwarte** [oddbird/popover-polyfill#183]
```js
if (element.popover === "auto") {
    const originalType = element.getAttribute("popover");
    const ancestor = topMostPopoverAncestor(element) || document2;
    hideAllPopoversUntil(ancestor, false, true);
    if (originalType !== element.getAttribute("popover") || !checkPopoverValidity(element, false)) {
      return;
    }
  }
```

The above snippet is part of the `showPopover` method in the polyfill (https://github.com/oddbird/popover-polyfill/blob/main/src/popover-helpers.ts#L268). When this is called inside a nested popover inside the shadow DOM, `hideAllPopoversUntil` incorrectly hides the parent popover, which doesn't happen in browsers that already support popover.

**Comment by jpzwarte** [comment]
This is the DOM structure
![CleanShot 2024-02-22 at 15 01 08](https://github.com/oddbird/popover-polyfill/assets/3968/72d98043-aa3f-4c4f-8a9a-181f585a5249)

**Comment by ddegan** [comment]
Hi! We are facing the same issue, since we're trying to include the polyfills for our Web Components. [Here](https://codesandbox.io/p/sandbox/festive-kapitsa-cnt6ql?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clto975h60006356hqlc9rrjp%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clto975h50002356hsly8q8h8%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clto975h50003356hs1j4g406%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clto975h50005356hadlq6g2m%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clto975h50002356hsly8q8h8%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clto975h50001356hiz6s45ny%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Findex.html%2522%257D%255D%252C%2522id%2522%253A%2522clto975h50002356hsly8q8h8%2522%252C%2522activeTabId%2522%253A%2522clto975h50001356hiz6s45ny%2522%257D%252C%2522clto975h50005356hadlq6g2m%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clto975h50004356hs29nes79%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clto975h50005356hadlq6g2m%2522%252C%2522activeTabId%2522%253A%2522clto975h50004356hs29nes79%2522%257D%252C%2522clto975h50003356hs1j4g406%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clto975h50003356hs1j4g406%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) a small example that reproduce the issue.

Thanks!

**Comment by jgerigmeyer** [comment]
@ddegan It looks like your case is not the same as the one described in this issue. The polyfill is running correctly in your code, but there's a race condition where the `defer` attribute on the `<script>` tag that loads the polyfill causes the polyfill to run _after_ `attachShadow` is called. The polyfill still "works", but this means the styles aren't be loaded in the shadow root. Either removing `defer` from the polyfill or adding `type="module"` to your `scripts.js` script tag fixes it. Here's a [forked version](https://codesandbox.io/p/sandbox/festive-kapitsa-forked-ldj3s3?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clto975h60006356hqlc9rrjp%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clto975h50002356hsly8q8h8%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clto975h50003356hs1j4g406%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clto975h50005356hadlq6g2m%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clto975h50002356hsly8q8h8%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clto975h50001356hiz6s45ny%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Findex.html%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clto975h50002356hsly8q8h8%2522%252C%2522activeTabId%2522%253A%2522clto975h50001356hiz6s45ny%2522%257D%252C%2522clto975h50005356hadlq6g2m%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clto975h50004356hs29nes79%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clto975h50005356hadlq6g2m%2522%252C%2522activeTabId%2522%253A%2522clto975h50004356hs29nes79%2522%257D%252C%2522clto975h50003356hs1j4g406%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clto975h50003356hs1j4g406%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) that removes `defer`.

**Comment by jpzwarte** [comment]
Thanks @jgerigmeyer !

**Comment by jgerigmeyer** [comment]
Included in [v0.4.1 release](https://github.com/oddbird/popover-polyfill/releases/tag/v0.4.1).
